src/Controller/GameController.php

```diff
@@ -17,12 +21,70 @@ public function __construct(GameManipulator $gameManipulator)
         $this->gameManipulator = $gameManipulator;
     }
 
-    public function random(): Response
-    {        
-        $game = $this->gameManipulator->createGame();
+    /**
+     * @ParamConverter("game", options={"mapping": {"hash": "hash"}})
+     */
+    public function random(Request $request, Game $game): Response
+    {
+        $triggers = [];
+
+        foreach (range(0, 9) as $x) {
+            foreach (range(0, 9) as $y) {
+                $triggers[] = $this->createForm(ShootShipType::class, [
+                    'game_hash' => $game->getHash(),
+                    'coordinates' => join(",", [$x, $y])
+                ]);
+            }
+        }
+
+        $trigger = $this->createForm(ShootShipType::class)->handleRequest($request);
+
+        if ($trigger->isSubmitted() && $trigger->isValid()) {
+            $data = $trigger->getData();
+
+            $game = $this->getDoctrine()
+                ->getRepository(Game::class)
+                ->findOneByHash($data['game_hash']);
+
+            if (!$game) {
+                throw $this->createNotFoundException(
+                    'No game found for hash '.$data['game_hash']
+                );
+            }
+            
+            $coordinates = explode(',', $data['coordinates']);
+
+            $this->gameManipulator->shoot($coordinates, $game);
+            
+            $this->gameManipulator->switchCurrentPlayer($game);
+
+            return $this->redirectToRoute('random_game', ['hash' => $game->getHash()]);
+        }
+
+        $triggerViews = [];
+        foreach ($triggers as $trigger) {
+            $triggerViews[] = $trigger->createView();
```

Comment:
Since you only need form views for the grid you can include it in the private method toi ;)